### PR TITLE
Fix #2807: Merge openmetadata.yaml and openmetadata-server.yaml

### LINKED ISSUE #2807: Merge openmetadata.yaml and openmetadata-server.yaml
**Feature**
Currently we keep non-secure configs in openmetadata.yaml and  secure configs openmetadata-server.yaml. 
Our ./bin/openmetadata-server-start.sh uses conf/openmetadata.yaml as the default config
It not only confuses our users but also hard to make the docker to go from non-secure to secure setup.

**Describe the task**
Have all the secure configs in openmetadata.yaml with authenticationConfiguration and authorizerConfiguration along with environment variables to secure those.
By default these will have non-secure configs. This will allow any interested users to continue to test OpenMetadata without securing through google or okta

### PR DESCRIPTION
**Pull request by harshach**
### Describe your changes :
see #2807 

#
### Type of change :

- [x] Improvement

#
### Frontend Preview (Screenshots) :
<p align="center">For frontend related change, please link screenshots of your changes preview! Optional for backend related changes.
</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [ ] I have performed a self-review of my own. 
- [ ] I have tagged my reviewers below.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

**Comment by harshach** [comment]
do not merge this until @Sachin-chaurasiya @darth-coder00 handles the UI code changes.
@Sachin-chaurasiya lets push the changes into same branch

**Comment by darth-coder00** [comment]
@harshach, handled UI for these yaml changes

**Comment by harshach** [comment]
@akash-jain-10 please review and merge it

File: conf/openmetadata.yaml [hunk 1]
```diff
@@ -113,6 +113,23 @@ database:
 migrationConfiguration:
   path: "./bootstrap/sql/mysql"
 
+# Authorizer Configuration
+authorizerConfiguration:
+  className: ${AUTHORIZER_CLASS_NAME:-org.openmetadata.catalog.security.NoopAuthorizer}
+  containerRequestFilter: ${AUTHORIZER_REQUEST_FILTER:-""}
```
Comment (mithmatt) [inline conf/openmetadata.yaml:119]: Should this be 


```
${AUTHORIZER_REQUEST_FILTER:-org.openmetadata.catalog.security.CatalogSecurityContextRequestFilter}
```

Comment (harshach) [reply conf/openmetadata.yaml]: Missed it, however it should be NoopFilter. Fixed it in latest commit

**Comment by harshach** [comment]
@darth-coder00 
if authorizerConfig is empty or if containerRequestFilter is empty or set to org.openmetadata.catalog.security.NoopFilter do not redirect to sign-in page

**Review by mithmatt — APPROVED** [review]

**Review by akash-jain-10 — APPROVED** [review]

**Pull request merged**